### DISCUSSION TITLE
Why doesn't Ghostty show my font?

### DISCUSSION THREAD

**Discussion by nathanmsmith** [ghostty-org/ghostty#4214] [Q&A] [answered]
I use the [Dank Mono](https://philpl.gumroad.com/l/dank-mono) font, but it doesn't show up when I run `ghostty +list-fonts`. Other terminal emulators like iTerm and Kitty pick up the font fine. How does Ghostty decide which fonts to support? Is there a way for it to use Dank Mono?

**Comment by rhodes-b** [comment]
what system are you on

**Reply by nathanmsmith** [reply]
macOS 15.2

**Reply by profbbrown** [reply]
I noticed this, too, with SF Mono Nerd Font. Even though it's configured as the terminal font, it doesn't show up in `+list-fonts`.

**Comment by ardupper** [comment]
Try going in to Font Book and checking if the Dank Mono font is part of the "Fixed Width" Smart Collection. I've struggled to find much good documentation of precisely *how* a font gets marked as fixed width, but on my system the list of fonts marked fixed width and the list of fonts under `ghostty +list-fonts` is identical.

Hope this helps

**Reply by ardupper** [reply]
macOS 15.1.1, btw

**Reply by nathanmsmith** [reply, marked as answer]
Dank Mono is not part of the "Fixed Width" collection! I went down a bit of a rabbit hole here – a [stack exchange user](https://apple.stackexchange.com/a/476917) thought that fixed width fonts are decided by PANOSE values.

I checked Dank Mono with the [ttx](https://fonttools.readthedocs.io/en/latest/ttx.html) utility and it had values of all 0. I edited this to be Family Kind=2, Proportion=9 (monospace), reinstalled the font, and restarted my computer. No luck.

Then after some more experimenting, I found that Ghostty does support the font, using the line:
```
font-family = Dank Mono
```

This is despite the fact that Dank Mono does not show up in `ghostty +list-fonts`. This appears to work for any font; if you love chaos try out `font-family = Comic Sans MS`. I'm not sure if this is intended behavior or not, the Ghostty config file implies that only fonts listed by  `+list-fonts` are valid.

**Reply by ardupper** [reply]
I bet it is intended behavior because there a features like smart kerning and `...` ligatures make them slightly fancier than a true "monospaced" font, but they're still perfect for terminal use. I had the same thoughts as you after reading the documentation.

**Comment by mohd-akram** [comment]
I don't see "SF Mono" when doing `ghostty +list-fonts`. I'm on macOS 15.2. This is all I get:

```
Andale Mono
  Andale Mono

Courier New
  Courier New
  Courier New Bold
  Courier New Bold Italic
  Courier New Italic

Menlo
  Menlo Bold
  Menlo Bold Italic
  Menlo Italic
  Menlo Regular

Monaco
  Monaco

PT Mono
  PT Mono
  PT Mono Bold
```

**Reply by lemomar** [reply]
Did you find a solution for this?

**Reply by leonasdev** [reply]
macOS do not have `SF Mono` by default. You need to download it from: https://developer.apple.com/fonts/

**Comment by magnusriga** [comment]
I am having the same problem, just with "JetBrainsMono Nerd Font".

It does not show up when running `+list-fonts`, and it also does not work when adding it to the config.

**Reply by rhodes-b** [reply]
This is the default font ghostty uses, it might not show up if its the baked in one but it will just JB Mono NF by default

**Reply by magnusriga** [reply]
Got it, that must be why I do not see a change.

Separately, but related:

One thing that seems odd is that on WezTerm, "JetBrainsMono Nerd Font" looks slightly more bold and less tall / lower line height. It might be because Ghostty forces the Mono version, "JetBrainsMono Nerd Font Mono", whereas WezTerm accepted the regular non-mono version, i.e. "JetBrainsMono Nerd Font".

I have the same font size, so I do not see why the two should look different (they are the same length).

WezTerm:

![image](https://github.com/user-attachments/assets/054d3f7b-5cff-4ee6-9949-db20cd021524)

Ghostty:

![image](https://github.com/user-attachments/assets/b23fa361-7908-4324-834c-f139e60f861d)

The difference is subtle, but the Ghostty version is taller / has more line spacing (or similar). It is quite noticeable on a screen full of text.

Anyone else seeing the same?

**Reply by rhodes-b** [reply]
Spacing would have to do with cell size likely which you can configure take a look at the website for more details ghostty.org
(on mobile otherwise id link myself)

**Reply by magnusriga** [reply]
I will check it out now. I noticed that WezTerm can currently fit 56 lines on one screen, while Ghostty fits 60. Same font-size. I will report back if cell size makes a difference. It does look like the characters themselves are slightly taller / fatter in WezTerm though. Not sure if cell size would affect the characters, I thought they came straight from the font, assume same size. I'll test further.

**Reply by magnusriga** [reply]
I now added `adjust-cell-height = 5%`, and tweaked padding, to get the two terminals as similar as possible. The thickness difference is now even clearer:

![gif-term-font](https://github.com/user-attachments/assets/a8b7d52c-ef9c-4935-8f6c-af290e06cc05)

I tried `font-thicken`, but that made the font way too thick.

Can it really be the same font? Each character is visibly taller in WezTerm.

As it currently stands, each character starts at the same top-line in  both terminals, but WezTerm just stretches a bit taller (without pushing the next line further down).

Thoughts or ideas appreciated :)

**Reply by rhodes-b** [reply]
yes, not all terminals render fonts the exact same way, you will notice differences in wezterm and kitty as well. thickness is not defined by a font rather the stroke of how characters look the characters look the same so yes its the same font

**Reply by VPraharsha03** [reply]
@rhodes-b 
I'm trying to use `JetBrainsMono Nerd Font Mono` in ghostty but it clearly looks like it's not being applied and ghostty continues to use it's own baked in font, it works perfectly on konsole & kitty - it's clear from the fact that `ghostty` is able to display the nobara's logo (because ghostty is using newer NF) but `kitty` & `konsole` are not able to display nobara's unicode as i'm using an older version of JBMono NF Mono

The issue i'm experiencing is the space after icon and text, on `ghostty` there is no space but on `kitty` and `konsole` spacing appears correctly... 

konsole:
![konsole.png](https://github.com/user-attachments/assets/577f7979-2918-4fa4-9077-5a8a01bbf9ab)

ghostty: 
![ghostty.png](https://github.com/user-attachments/assets/903168ed-0896-4253-a63c-46d0fcf111fc)

ghostty config:
![ghostty_config.png](https://github.com/user-attachments/assets/8674c96a-5b7c-427c-8af1-7b34612acdec)

**Reply by danielwe** [reply]
> `ghostty` is able to display the nobara's logo (because ghostty is using newer NF)

This is due to fallbacks. Ghostty has a built-in font that provides this icon, so it will always be able to display it, no matter what your primary font is. This isn't just for nerd fonts---if your primary font doesn't provide a codepoint, Ghostty will always do its best to find a fallback font that does.

> The issue i'm experiencing is the space after icon and text, on `ghostty` there is no space

What version of Ghostty are you on? If you upgrade to the most recent release, 1.2.2, I think this should be fixed.

**Reply by VPraharsha03** [reply]
I'm on 1.2.0, will try upgrading and see

EDIT: upgraded to 1.2.2 and issue got solved

![ghostty_updated.png](https://github.com/user-attachments/assets/b4d5be07-9d60-4b96-830e-c9047b169127)

**Comment by mariansam** [comment]
If anyone comes here because Ghostty (on Linux) opens without rendering anything and `ghostty +list-fonts` returns nothing: run `fc-cache -fv` to rebuild the font cache and try again.
